src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -2019,7 +2028,91 @@ await PreviewBuildIntegratedProjectActionsAsync(buildIntegratedProject, actions,
                 throw new ArgumentNullException(nameof(nuGetProjectContext));
             }
 
+            token.ThrowIfCancellationRequested();
+
+            var results = new List<NuGetProjectAction>();
+
+            // BuildIntegratedNuGetProject type projects are now supports parallel preview action for faster performance.
+            var buildIntegratedProjectsToUpdate = new List<BuildIntegratedNuGetProject>();
+            var otherTargetProjectsToUpdate = new List<NuGetProject>();
+
+            foreach (var proj in nuGetProjects)
```

Comment:
You can make this one line with a where statement + ToList.

Reply:
I know. If I remember correctly previously I received code review stating this one is more performant than that one.